src/PlaywrightSharp.Tests/Evaluation/FrameEvaluateTests.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System.Threading.Tasks;
+using PlaywrightSharp.Tests.BaseTests;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace PlaywrightSharp.Tests.Evaluation
+{
+    ///<playwright-file>evaluation.spec.js</playwright-file>
+    ///<playwright-describe>Frame.evaluate</playwright-describe>
+    public class FrameEvaluateTests : PlaywrightSharpPageBaseTest
+    {
+        internal FrameEvaluateTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        ///<playwright-file>evaluation.spec.js</playwright-file>
+        ///<playwright-describe>Frame.evaluate</playwright-describe>
+        ///<playwright-it>should have different execution contexts</playwright-it>
+        [Fact]
+        public async Task ShouldHaveDifferentExecutionContexts()
+        {
+            await Page.GoToAsync(TestConstants.EmptyPage);
+            await FrameUtils.AttachFrameAsync(Page, "frame1", TestConstants.EmptyPage);
+            Assert.Equal(2, Page.Frames.Length);
+            await Page.Frames[0].EvaluateAsync("() => window.FOO = 'foo'");
+            await Page.Frames[1].EvaluateAsync("() => window.FOO = 'bar'");
+            Assert.Equal("foo", await Page.Frames[0].EvaluateAsync<string>("() => window.FOO"));
+            Assert.Equal("bar", await Page.Frames[1].EvaluateAsync<string>("() => window.FOO"));
+        }
+
+        ///<playwright-file>evaluation.spec.js</playwright-file>
+        ///<playwright-describe>Frame.evaluate</playwright-describe>
+        ///<playwright-it>should have correct execution contexts</playwright-it>
+        [Fact]
+        public async Task ShouldHaveCorrectExecutionContexts()
+        {
+            await Page.GoToAsync(TestConstants.ServerUrl + "/frames/one-frame.html");
+            Assert.Equal(2, Page.Frames.Length);
+            Assert.Empty(await Page.Frames[0].EvaluateAsync<string>("() => document.body.textContent.trim()"));
+            Assert.Equal("Hi, I'm frame", await Page.Frames[1].EvaluateAsync<string>("() => document.body.textContent.trim()"));
+        }
+
+        ///<playwright-file>evaluation.spec.js</playwright-file>
+        ///<playwright-describe>Frame.evaluate</playwright-describe>
+        ///<playwright-it>should dispose context on navigation</playwright-it>
+        [Fact]
+        public async Task ShouldDisposeContextOnNavigation()
+        {
+            await Page.GoToAsync(TestConstants.ServerUrl + "/frames/one-frame.html");
+            Assert.Equal(2, Page.Frames.Length);
+            // TODO: expect(page._delegate._contextIdToContext.size).toBe(4);

```

Comment:
_delegate => `Delegate` can be an internal `IPageDelegate` property.
_contextIdToContext => ContextIdToContext and internal `Dictionary<int, IExecutionContext>`

Reply:
I just realized we can't do this with our interfaces, can't we do an Assert.Fail so we don't forget about these To-Dos